django/db/models/query.py

```diff
@@ -1286,40 +1336,70 @@ def db(self):
     # PRIVATE METHODS #
     ###################
 
-    def _insert(self, objs, fields, returning_fields=None, raw=False, using=None, ignore_conflicts=False):
+    def _insert(
+        self, objs, fields, returning_fields=None,
+        raw=False, using=None, on_conflict=None, update_fields=None,
+        unique_fields=None
+    ):
         """
         Insert a new record for the given model. This provides an interface to
         the InsertQuery class and is how Model.save() is implemented.
         """
         self._for_write = True
         if using is None:
             using = self.db
-        query = sql.InsertQuery(self.model, ignore_conflicts=ignore_conflicts)
+        query = sql.InsertQuery(
+            self.model, on_conflict=on_conflict, update_fields=update_fields,
+            unique_fields=unique_fields
+        )
         query.insert_values(fields, objs, raw=raw)
         return query.get_compiler(using=using).execute_sql(returning_fields)
     _insert.alters_data = True
     _insert.queryset_only = False
 
-    def _batched_insert(self, objs, fields, batch_size, ignore_conflicts=False):
+    def _check_on_conflicts_supported(self, on_conflict, unique_fields):
+        db_features = connections[self.db].features
+        if on_conflict == OnConflict.Ignore and not db_features.supports_ignore_conflicts:
+            raise NotSupportedError('This database backend does not support ignoring conflicts.')
+        if on_conflict == OnConflict.Update:
+            if unique_fields and not db_features.supports_update_conflicts_with_target:
+                raise NotSupportedError(
+                    'This database backend does not support updating conflicts with '
+                    'specifying unique fields that will trigger the upsert.'
+                )
+            elif not db_features.supports_update_conflicts:
+                raise NotSupportedError(
+                    'This database backend does not support updating conflicts.'
+                )
+
+    def _batched_insert(
+        self, objs, fields, batch_size,
+        on_conflict=None, update_fields=None, unique_fields=None
```

Comment:
Please add a trailing comma:
```suggestion
        on_conflict=None, update_fields=None, unique_fields=None,
```